packages/hoppscotch-common/src/services/context-menu/index.ts

```diff
@@ -0,0 +1,95 @@
+import { Service } from "dioc"
+import { Component } from "vue"
+
+/**
+ * Defines how to render the text in a Context Menu Search Result
+ */
+export type ContextMenuTextType<T extends object | Component = never> =
+  | {
+      type: "text"
+      text: string
+    }
+  | {
+      type: "custom"
+      /**
+       * The component to render in place of the text
+       */
+      component: T
+
+      /**
+       * The props to pass to the component
+       */
+      componentProps: T extends Component<infer Props> ? Props : never
+    }
+
+/**
+ * Defines info about a context menu result so the UI can render it
+ */
+export interface ContextMenuResult {
+  /**
+   * The unique ID of the result
+   */
+  id: string
+  /**
+   * The text to render in the result
+   */
+  text: ContextMenuTextType<any>
+  /**
+   * The icon to render as the signifier of the result
+   */
+  icon: object | Component
+  /**
+   * The action to perform when the result is selected
+   */
+  action: () => void
+  /**
+   * Additional metadata about the result
+   */
+  meta?: {
+    /**
+     * The keyboard shortcut to trigger the result
+     */
+    keyboardShortcut?: string[]
+  }
+}
+
+/**
+ * Defines the state of a context menu
+ */
+export type ContextMenuState = {
+  results: ContextMenuResult[]
+}
+
+/**
+ * Defines a context menu
+ */
+export interface ContextMenu {
```

Comment:
This should be a `type` rather than an `interface` right ?

```suggestion
export type ContextMenu = {
```